examples/admin.js

```diff
@@ -0,0 +1,162 @@
+const fs = require('fs')
+const ip = require('ip')
+
+const { Kafka, logLevel } = require('../index')
+
+const host = process.env.HOST_IP || ip.address()
+
+const logCreator = _logLevel => {
+  return ({ namespace, level, label, log }) => {
+    const { timestamp, message, error, ...extra } = log
+    console.log(`${timestamp}: ${label}: ${message} ${error}`)
+  }
+}
+
+const kafka = new Kafka({
+  logLevel: logLevel.INFO,
+  brokers: [`${host}:9094`, `${host}:9097`, `${host}:9100`],
+  clientId: 'test-admin-id',
+  ssl: {
+    servername: 'localhost',
+    rejectUnauthorized: false,
+    ca: [fs.readFileSync('./testHelpers/certs/cert-signed', 'utf-8')],
+  },
+  sasl: {
+    mechanism: 'plain',
+    username: 'test',
+    password: 'testtest',
+  },
+})
+
+const topic1 = 'topic-test1'
+const topic2 = 'topic-test2'
+const topic3 = 'topic-test3'
+
+const expectedTopics = [topic1, topic2, topic3]
+
+const filterTopics = ({ name }) => expectedTopics.includes(name)
+const sortTopics = (a, b) => {
+  if (a.name < b.name) {
+    return -1
+  }
+  if (a.name > b.name) {
+    return 1
+  }
+  return 0
+}
+
+const admin = kafka.admin()
+
+const getMyTopics = async () => {
+  const md = await admin.fetchTopicMetadata({})
+  return md.topics.filter(filterTopics).sort(sortTopics)
+}
+
+const wait = nrOfSec => {
+  return new Promise((resolve, _) => {
+    setTimeout(resolve, nrOfSec * 1000)
+  })
+}
+
+const run = async () => {

```

Comment:
I think this file should not be part of master. The examples in here should be as minimal as possible, and show a simple version of a common task (produce a few messages, consume from a topic). This is way too detailed and will be hard for people to understand.